modules/ppcp-button/resources/js/modules/ContextBootstrap/CheckoutBootstap.js

```diff
@@ -100,13 +102,45 @@ class CheckoutBootstap {
             this.renderer.hideButtons(this.gateway.messages.wrapper)
             this.renderer.hideButtons(this.gateway.hosted_fields.wrapper)
             jQuery('#place_order').show()
+            this.disableCreditCardFields()
         } else {
             jQuery('#place_order').hide()
             this.renderer.hideButtons(this.gateway.button.wrapper)
             this.renderer.hideButtons(this.gateway.messages.wrapper)
             this.renderer.showButtons(this.gateway.hosted_fields.wrapper)
+            this.enableCreditCardFields()
         }
     }
+
+    disableCreditCardFields() {
+        jQuery('label[for="ppcp-credit-card-gateway-card-number"]').css({ opacity: 0.4 })
+        jQuery('#ppcp-credit-card-gateway-card-number').css({ opacity: 0.4 })
+        jQuery('#ppcp-credit-card-gateway-card-number').attr("disabled", true);
+        jQuery('label[for="ppcp-credit-card-gateway-card-expiry"]').css({ opacity: 0.4 })
+        jQuery('#ppcp-credit-card-gateway-card-expiry').css({ opacity: 0.4 })
+        jQuery('#ppcp-credit-card-gateway-card-expiry').attr("disabled", true)
+        jQuery('label[for="ppcp-credit-card-gateway-card-cvc"]').css({ opacity: 0.4 })
+        jQuery('#ppcp-credit-card-gateway-card-cvc').css({ opacity: 0.4 })
+        jQuery('#ppcp-credit-card-gateway-card-cvc').attr("disabled", true)
+        jQuery('label[for="vault"]').css({ opacity: 0.4 })
+        jQuery('#ppcp-credit-card-vault').css({ opacity: 0.4 })
```

Comment:
Maybe for opacity it would be better and cleaner to add a class via JS and specify these styles in a CSS file.  Also I think it is enough to specify opacity just for their parent container?

Reply:
I tried adding a css class with opacity via JS but it´s overriden by the inline styles:
![hosted-css](https://user-images.githubusercontent.com/456223/133051259-4df1cf56-a6ee-4dc4-96a6-3686343ef7c2.png)

Seems adding the property via JS (using jQuery in this case) overwrites the inline styles.

Reply:
Done.

Reply:
hm, the class name is strange, missing `-disabled` like on the image above.

Also you can probably just add these classes once on init and then only set `disabled` attribute and use `myclass:disabled` selector in CSS. https://developer.mozilla.org/en-US/docs/Web/CSS/:disabled

Reply:
`-disabled` added to class name. Regarding css `disabled` attribute, I´m also using the class on labels which are not disabled. I would prefer to use only one approach for both (labels and inputs) if possible.